channel/contractbackend.go

```diff
@@ -165,10 +187,7 @@ func (c *ContractBackend) nonce(ctx context.Context, sender common.Address) (uin
 		err = cherrors.CheckIsChainNotReachableError(err)
 		return 0, errors.WithMessage(err, "fetching nonce")
 	}
-
-	// Look up expected next nonce locally.
-	c.nonceMtx.Lock()
-	defer c.nonceMtx.Unlock()
+	SharedMutex.Lock()
```

Comment:
The previous defer Unlock is probably more idiomatic Go (as it enforces the unlock should a panic happen, too).

Though as I don't think the code in between can panic (besides an out of memory error) they should be functionally equivalent. I don't know which one you usually use in perun code.